pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -748,10 +730,10 @@ def _update_best_and_save(
                 f"Epoch {epoch:d}, global step {step:d}: {self.monitor} reached {current:0.5f}"
                 f' (best {self.best_model_score:0.5f}), saving model to "{filepath}" as top {k}'
             )
-        self._save_model(trainer, filepath)
+        trainer.save_checkpoint(filepath, self.save_weights_only)
 
         if del_filepath is not None and filepath != del_filepath:
-            self._del_model(trainer, del_filepath)
+            trainer.training_type_plugin.remove_checkpoint(del_filepath)
```

Comment:
do we care about routing through the accelerator for these functions? are we ok with calling the training_type_plugin directly?

Reply:
i'm in favor of this as the direction is to flip the accelerator & training type relationship. 

making the accelerator be part of the training type plugin to manage device-specific logic folds in as part of the overall parallelization strategy. 

an incremental approach would be to:
- update all calls in the framework that currently route from accelerator -> training type plugin to call the training type directly
- remove those APIs from the accelerator
- then ultimately flip the nesting

cc @four4fish

Reply:
created an issue here to discuss this further: https://github.com/PyTorchLightning/pytorch-lightning/issues/9426